+sw_tests/+unit_tests/unittest_spinw_addatom.m

```diff
@@ -0,0 +1,110 @@
+classdef unittest_spinw_addatom < matlab.mock.TestCase
+
+    properties
+        swobj = [];
+    end
+
+    methods (TestMethodSetup)
+        function setup_spinw_model(testCase)
+            testCase.swobj = spinw(); % default init
+        end
+    end
+
+    methods (Test)
+        function test_no_input_calls_help(testCase)
+            % Tests that if call addmatrix with no input, it calls the help
+            help_function = sw_tests.utilities.mock_function('swhelp');
+            testCase.swobj.addatom();
+            testCase.assertEqual(help_function.n_calls, 1);
+            testCase.assertEqual(help_function.arguments, ...
+                {{'spinw.addatom'}});
+        end
+        
+        function test_add_multiple_atom_throws_mismatch_spin(testCase)
+            testCase.verifyError(...
+                @() testCase.swobj.addatom('r', [0 0;0 0.5;0 0.5], 'S', 1), ...
+                'spinw:sw_valid:SizeMismatch')
+        end
+        
+        function test_add_atom_fails_invalid_position_size(testCase)
+            testCase.verifyError(...
+                @() testCase.swobj.addatom('r', [0, 0]), ...
+                'MATLAB:NonIntegerInput')  % need a better error for this
+        end
+        
+        function test_add_atom_fails_without_position(testCase)
+            testCase.verifyError(...
+                @() testCase.swobj.addatom('S', 1), ...
+                'sw_readparam:MissingParameter')
+        end
+        
+        function test_add_atom_fails_with_negative_spin(testCase)
+            testCase.verifyError(...
+                @() testCase.swobj.addatom('r', [0; 0; 0], 'S', -1), ...
+                'spinw:addatom:WrongInput')
+        end
+        
+        function test_add_single_default_atom_with_only_position(testCase)
+            pos = [0; 0; 0];
+            testCase.swobj.addatom('r', pos)
+            unit_cell = testCase.swobj.unit_cell;
+            testCase.assertEqual(unit_cell.r, pos)
+            testCase.assertEqual(unit_cell.S, 0)  % default non-mag
+            testCase.assertEqual(unit_cell.A, int32(-1))  % natural isotope mix
+            testCase.assertEqual(unit_cell.Z, int32(113))  % same for all subsequent default atoms added
```

Comment:
Should this be changed to `0` or `-1`...?

[Actually... scanning through the code it seems that the atomic number `Z` is not actually used anywhere... the code which sets it says that Z=113 is "Unobtanium" but this is outdated... element 113 is Nihonium and at least seems to have been obtained by researchers at Riken... If we want to keep with the times we should update Unobtanium to Z=119 ;]

Reply:
Yes, perhaps I should remove the comment entirely! We could make the default `Z` it `0`, or `-1` - I think this will break the lookups for scattering length etc. - however in this case why not force the user to supply a valid atom/Z ?

Reply:
Apparently there is a quick check of a matrix is symmetric positive definite (I think g needs to be symmetric - i.e. `g_ij = g_ji`) - we can check the `flag` (optional output) of the `chol` function. Just depends on whether we should allow zero eigenvalues...
According to section D in this paper
https://journals.aps.org/prx/pdf/10.1103/PhysRevX.8.021068
the matrix `g^\dagger \cdot g` is symmetric, positive definite

Reply:
Thanks for looking this up... I checked [Abragam and Bleaney](https://books.google.co.uk/books?id=ASNoAgAAQBAJ&lpg=PA652&pg=PA652#v=onepage&q&f=false) and actually the necessary condition is:

![AbragamBleaney_gtensor](https://latex.codecogs.com/png.image?\dpi{110}G_{pq}&space;=&space;\sum_{\alpha}&space;g_{p\alpha}&space;g_{q\alpha})

and it is **Gij** which needs to be positive definite rather than the original "g-tensor". This is because (as Abragam and Bleaney explain) the **Gij** come from the expressions for the eigenvalues of the squared Zeeman Hamiltonian which must be positive and real.

Reply:
Ah sorry I clicked the wrong box - the comment above should go with the next bit about `addg`.

Anyway, about the default values, I think we should just leave it at `Z=113` for now.